Modules/Scripted/Endoscopy/Endoscopy.py

```diff
@@ -707,28 +551,31 @@ class EndoscopyLogic:
 
     NODE_PATH_CAMERA_ORIENTATIONS_ATTRIBUTE_NAME = "Endoscopy.Path.CameraOrientations"
 
-    def __init__(self, inputCurve, cameraOrientations, dl=0.5):
-        self.cleanup()
-        self.cameraOrientations = cameraOrientations
+    def __init__(self, dl=0.5):
         self.dl = dl  # desired world space step size (in mm)
-        self.setControlPoints(inputCurve)
-
-    def cleanup(self):
-        # Whether we're about to construct or delete, free all resources and initialize class members to None.
-        self.dl = None
         self.inputCurve = None
-        self.resampledCurve = None
+
         self.planeNormal = None
-        self.cameraOrientations = None
+
+        self.resampledCurve = slicer.vtkMRMLMarkupsCurveNode()
         self.cameraOrientationResampledCurveIndices = None
 
-    def setControlPoints(self, inputCurve: slicer.vtkMRMLMarkupsCurveNode) -> bool:
+        self.updatingControlPoints = False
+
+    def getNumberOfControlPoints(self):
+        return self.resampledCurve.GetNumberOfControlPoints()
+
+    def setControlPoints(self, inputCurve: slicer.vtkMRMLMarkupsCurveNode) -> None:
```

Comment:
`getNumberOfControlPoints` is talking about `self.resampledCurve` but `setControlPoints` takes as an argument what will soon be `self.inputCurve`.  It's a little bit confusing, so I'd add a comment to the effect that `setControlPoints` gets its name from the fact that it using the control points of `self.inputCurve` to compute control points for `self.resampledCurve`.

Reply:
Perhaps we could change the name of the method from `setControlPoints` to `computeControlPoints`.

Reply:
That could work, though "update from" could mean "copy from", so I was aiming for a word like "compute", "calculate", "infer", "determine", ....

Reply:
Good idea.

There is a precedent for both `calculate` and `compute`, I will go with `calculate`

```$ ack --cpp -i "\:calculate.+From"
Modules/Loadable/Markups/MRML/vtkMRMLMarkupsPlaneNode.cxx
563:void vtkMRMLMarkupsPlaneNode::CalculateAxesFromPoints(const double point0[3], const double point1[3], const double point2[3],

Modules/Loadable/Segmentations/Logic/vtkSlicerSegmentationGeometryLogic.cxx
178:std::string vtkSlicerSegmentationGeometryLogic::CalculateOutputGeometryFromImage()
262:std::string vtkSlicerSegmentationGeometryLogic::CalculateOutputGeometryFromBounds(bool keepCurrentAxisDirections)
```

```$ ack --cpp -i "\:compute.+From"
Libs/MRML/Core/Testing/vtkMRMLVolumeNodeTest1.cxx
84:    vtkMRMLVolumeNode::ComputeIJKToRASFromScanOrder(
89:        vtkMRMLVolumeNode::ComputeScanOrderFromIJKToRAS(ijkToRAS.GetPointer());

Libs/MRML/Core/vtkMRMLVolumeNode.cxx
510:bool vtkMRMLVolumeNode::ComputeIJKToRASFromScanOrder(const char *order,
614:const char* vtkMRMLVolumeNode::ComputeScanOrderFromIJKToRAS(vtkMatrix4x4 *ijkToRAS)

Libs/MRML/Widgets/qMRMLVolumeInfoWidget.cxx
249:    vtkMRMLVolumeNode::ComputeScanOrderFromIJKToRAS(mat.GetPointer())));
363:  if (vtkMRMLVolumeNode::ComputeIJKToRASFromScanOrder(
```

Reply:
I ended up going with the explicit name of `resampleAndInterpolateControlPointsFromInputCurve`

Reply:
Similarly, I think we should rename `interpolateOrientations` to `interpolateControlPointsWithCameraOrientations` ?

Reply:
That's a good one, but I wouldn't start with "interpolate control points" because that is too easy to confuse with what `resampleAndInterpolateControlPointsFromInputCurve` does.  How about `interpolateCameraOrientationsBetweenKeyframes`?

Reply:
Since the concept of `KeyFrame` is not part of the API and what we consider keyframes are in fact labeled as `CameraOrientations` or `Orientations`, I am not sure we want to use `KeyFrames` in the context of `EndoscopyLogic`. This terminology is currently only associated with the user interface.

Since we are literally interpolating between all control points[^1] of the flythrough path (internally represented by `resampledCurve` internal variable), the name `interpolateControlPointsWithOrientations`  nicely capture that and is consistent with:
* `getNumberOfControlPoints`
* `resampleAndInterpolateControlPointsFromInputCurve`
* `saveOrientationAtIndex`
* `removeOrientationAtIndex`
* `updateCameraFromOrientationAtIndex`

Since we are interpolating between all control points while accounting for the _Orientation_ there are either the default ones (or the default ones combined with the ones that were saved) that seems reasonable.

Using `Orientation` (instead of `CameraOrientation`) is also consistent.

[^1]: Resampled control points are also technically keyframes that we are interpolates.